sql-plugin/src/main/spark320/scala/org/apache/spark/sql/rapids/shims/datetimeExpressions.scala

```diff
@@ -164,10 +164,56 @@ case class GpuTimeAdd(start: Expression,
     }
   }
 
+  // A tricky way to check overflow. The result is overflow when positive + positive = negative
+  // or negative + negative = positive, so we can check the sign of the result is the same as
+  // the sign of the operands.
   private def timestampAddDuration(cv: ColumnView, duration: BinaryOperable): ColumnVector = {
     // Not use cv.add(duration), because of it invoke BinaryOperable.implicitConversion,
     // and currently BinaryOperable.implicitConversion return Long
     // Directly specify the return type is TIMESTAMP_MICROSECONDS
-    cv.binaryOp(BinaryOp.ADD, duration, DType.TIMESTAMP_MICROSECONDS)
+    val resWithOverflow = cv.binaryOp(BinaryOp.ADD, duration, DType.TIMESTAMP_MICROSECONDS)
+    closeOnExcept(resWithOverflow) { _ =>
```

Comment:
Same comment as above, except at a minimum the common code should be moved to a utility that is not shimmed so we don't have to maintain it twice.